src/transformState.js

```diff
@@ -5,7 +5,30 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const el of actions) {
+    switch (el.type) {
+      case 'addProperties':
+        Object.assign(state, el.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const keysRemove of el.keysToRemove) {
+          delete state[keysRemove];
+        };
+        break;
+
+      case 'clear':
+        for (const key in state) {
+          if (state.hasOwnProperty([key])) {
+            delete state[key];
+          }
```

Comment:
you don't need `hasOwnProperty` here, delete will ignore keys which doesn't exist